File: Java/Structural/FacadePattern.java
package Structural;

// Subsystems
class Amplifier {
    public void on() { System.out.println("Amplifier on."); }
    public void off() { System.out.println("Amplifier off."); }
}

class Projector {
    public void on() { System.out.println("Projector on."); }
    public void off() { System.out.println("Projector off."); }
}

class DVDPlayer {
    public void on() { System.out.println("DVD Player on."); }
    public void off() { System.out.println("DVD Player off."); }
}

// Facade
class HomeTheaterFacade {
    private Amplifier amplifier;
    private Projector projector;
    private DVDPlayer dvdPlayer;
    
    public HomeTheaterFacade(Amplifier amplifier, Projector projector, DVDPlayer dvdPlayer) {
        this.amplifier = amplifier;
        this.projector = projector;
        this.dvdPlayer = dvdPlayer;
    }
    
    public void watchMovie() {
        amplifier.on();
        projector.on();
        dvdPlayer.on();
        System.out.println("Movie is now playing!");
    }
    
    public void endMovie() {
        amplifier.off();
        projector.off();
        dvdPlayer.off();
        System.out.println("Movie ended.");
    }
}

File: Python/Structural/FacadePattern.py
# Subsystems
class Amplifier:
    def on(self):
        print("Amplifier on")

    def off(self):
        print("Amplifier off")

class Projector:
    def on(self):
        print("Projector on")

    def off(self):
        print("Projector off")

class DVDPlayer:
    def on(self):
        print("DVD Player on")

    def off(self):
        print("DVD Player off")

# Facade
class HomeTheaterFacade:
    def __init__(self, amplifier, projector, dvd_player):
        self.amplifier = amplifier
        self.projector = projector
        self.dvd_player = dvd_player

    def watch_movie(self):
        print("Get ready to watch a movie...")
        self.amplifier.on()
        self.projector.on()
        self.dvd_player.on()

    def end_movie(self):
        print("Shutting down home theater...")
        self.amplifier.off()
        self.projector.off()
        self.dvd_player.off()

# Client usage
amplifier = Amplifier()
projector = Projector()
dvd_player = DVDPlayer()

home_theater = HomeTheaterFacade(amplifier, projector, dvd_player)
home_theater.watch_movie()
home_theater.end_movie()
